File: threads/src/main/java/im/threads/internal/model/ConsultTyping.java
package im.threads.internal.model;

import androidx.core.util.ObjectsCompat;

import im.threads.business.models.ChatItem;
import im.threads.business.models.ConsultChatPhrase;

public final class ConsultTyping extends ConsultChatPhrase implements ChatItem {
    private long date;

    @Override
    public long getTimeStamp() {
        return date;
    }

    public ConsultTyping(String consultId, long date, String avatarPath) {
        super(avatarPath, consultId);
        this.date = date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public long getDate() {
        return date;
    }

    @Override
    public boolean isTheSameItem(ChatItem otherItem) {
        return otherItem instanceof ConsultTyping;
    }

    @Override
    public Long getThreadId() {
        return null;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ConsultTyping that = (ConsultTyping) o;
        return date == that.date;
    }

    @Override
    public int hashCode() {
        return ObjectsCompat.hash(date);
    }
}

File: src/main/java/im/threads/business/models/ConsultTyping.kt
package im.threads.business.models

import androidx.core.util.ObjectsCompat

class ConsultTyping(consultId: String?, override var timeStamp: Long, avatarPath: String?) :
    ConsultChatPhrase(avatarPath, consultId), ChatItem {
    override fun isTheSameItem(otherItem: ChatItem?): Boolean {
        return otherItem is ConsultTyping
    }

    override val modified = null
    override val threadId: Long?
        get() = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ConsultTyping
        return timeStamp == that.timeStamp
    }

    override fun hashCode(): Int {
        return ObjectsCompat.hash(timeStamp)
    }
}
